feature/metis/shared/src/main/kotlin/de/tum/informatics/www1/artemis/native_app/feature/metis/shared/ui/profile_picture/ProfilePicture.kt

```diff
@@ -0,0 +1,140 @@
+                ProfilePictureImage(
+                    modifier = modifier,
+                    profilePictureData = profilePictureData,
+                    profilePictureImageProvider = profilePictureImageProvider,
+                )
+            } else {
+                InitialsPlaceholder(
+                    modifier = modifier,
+                    profilePictureData = profilePictureData.fallBack,
+                )
+            }
+        }
+        is ProfilePictureData.InitialsPlaceholder -> {
+            InitialsPlaceholder(
+                modifier = modifier,
+                profilePictureData = profilePictureData,
+            )
+        }
+    }
+}
+
+@Composable
+fun ProfilePictureImage(
+    modifier: Modifier,
+    profilePictureData: ProfilePictureData.Image,
+    profilePictureImageProvider: ProfilePictureImageProvider,
+) {
+    val imageUrl = profilePictureData.url
+    val context = LocalContext.current
+    val request = remember(imageUrl, profilePictureImageProvider) {
+        profilePictureImageProvider.createImageRequest(
+            context = context,
+            imagePath = imageUrl,
+        )
+    }
+
+    val resultData = loadAsyncImageDrawable(request = request)
+
+    when (resultData.dataState) {
+        is DataState.Failure -> {
+            InitialsPlaceholder(
+                modifier = modifier,
+                profilePictureData = profilePictureData.fallBack,
+            )
+        }
+        is DataState.Loading -> {
+            InitialsPlaceholder(
+                modifier = modifier,
+                profilePictureData = profilePictureData.fallBack,
+            )
+        }
+        is DataState.Success -> {
+            val loadedDrawable = (resultData.dataState as DataState.Success<Drawable>).data
+
+            val painter = remember(loadedDrawable) {
+                val bitmap = loadedDrawable.toBitmap().asImageBitmap()
+                BitmapPainter(bitmap)
+            }
+
+            Image(
+                modifier = modifier
+                    .fillMaxWidth()
+                    .testTag(TEST_TAG_PROFILE_PICTURE_IMAGE),
+                painter = painter,
+                contentDescription = null,
+                contentScale = ContentScale.Fit
+            )
+        }
+    }
+}
+
+@Composable
+fun InitialsPlaceholder(
+    modifier: Modifier,
+    profilePictureData: ProfilePictureData.InitialsPlaceholder,
+) {
+    val boxSize = remember { mutableIntStateOf(0) }
+
+    Box(
+        modifier = modifier
+            .background(color = profilePictureData.backgroundColor)
+            .onGloballyPositioned {
+                boxSize.intValue = it.size.width
+            }
+            .testTag(TEST_TAG_PROFILE_PICTURE_INITIALS),
+        contentAlignment = Alignment.Center,
+    ) {
+        Text(
+            text = profilePictureData.initials,
+            color = Color.White,
+            fontSize = boxSize.intValue.sp.nonScaledSp * 0.16f,
```

Comment:
Where does the magic number 0.16 come from? Please declare it as a file constant.